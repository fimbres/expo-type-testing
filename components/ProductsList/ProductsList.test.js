import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';

import { ProductsList } from '.';

const testProducts = [
    {
        createdAt: "2022-12-08T18:54:56.243Z",
        id: 5,
        image: "https://loremflickr.com/640/480/people",
        is_redemption: true,
        points: 69814,
        product: "Rustic Rubber Bacon",
    },
    {
        createdAt: "2022-12-08T18:54:56.243Z",
        id: 5,
        image: "https://loremflickr.com/640/480/people",
        is_redemption: true,
        points: 69814,
        product: "Rustic Rubber Bacon",
    },
    {
        createdAt: "2022-12-08T18:54:56.243Z",
        id: 5,
        image: "https://loremflickr.com/640/480/people",
        is_redemption: true,
        points: 69814,
        product: "Rustic Rubber Bacon",
    }
];

describe('<ProductList />', () => {
    it('render empty state correctly', () => {
        render(
            <ProductsList
                product={[]}
                onPress={(id) => console.log("onPress", id)}
            />
        );

        const primaryText = screen.getByText('No hay movimientos registrados');
        const secondaryText = screen.getByText('Realiza una compra y vuelve mÃ¡s tarde');

        expect(primaryText).toBeDefined();
        expect(secondaryText).toBeDefined();
    });

    it('renders correctly', () => {
        const tree = renderer
        .create(
            <ProductsList
                product={testProducts}
                onPress={(id) => console.log("onPress", id)}
            />
        ).toJSON();

        expect(tree.children.length).toBe(1);
        expect(tree).toMatchSnapshot();
    });
});
