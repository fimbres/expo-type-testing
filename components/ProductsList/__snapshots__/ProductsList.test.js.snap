// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> renders correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 16,
      "flex": 1,
      "paddingVertical": 23,
      "width": "100%",
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 25,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 150,
            "marginBottom": 24,
            "resizeMode": "cover",
            "width": 190,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#020202",
            "fontFamily": "avenir-black",
            "fontSize": 16,
            "fontWeight": "800",
            "lineHeight": 21.86,
          }
        }
      >
        No hay movimientos registrados
      </Text>
      <Text
        style={
          Object {
            "color": "#9B9898",
            "fontFamily": "avenir-book",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 21.86,
          }
        }
      >
        Realiza una compra y vuelve m√°s tarde
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "marginTop": 33,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
